### PR TITLE
Expose and utilize pan and tilt faults in PLEXIL

### PR DESCRIPTION
**Pull request by pthangeda**
## Linked Issues:
| EPIC ⚡ | N/A |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-779](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-779) |
| Github :octocat:  | # |

## Summary of Changes
* New Lookups `AntennaPanFault` and `AntennaTiltFault` in addition to the existing `AntennaFault`.
* New  test plan `TestFaults.plp`

## Test
* Fetch the `OCEANWATER-934_telemetry_unification_epic` branch of `ow_simulator` to accompany this branch.
* Clean build, resource setup files.
* Start any simulator world.
* In other terminal, run `roslaunch ow_plexil ow_exec.launch`
* Run the plan `TestFaults.plx`
* Inject and clear antenna pan and tilt joint locked faults in any order, and see that the ROS messages printed in the ow_plexil terminal are accurate.  An example sequence is as follows.

```
[ INFO]: PLEXIL: Inject faults and observe messages. Type Control-C to exit plan and exec node.
[ INFO]: PLEXIL: Antenna is fine.
[ INFO]: PLEXIL: Pan is fine.
[ INFO]: PLEXIL: Tilt is fine.
[ INFO]: PLEXIL: Antenna is fine.
[ INFO]: PLEXIL: Pan is fine.
[ INFO]: PLEXIL: Tilt is fine.
[ INFO]: PLEXIL: Antenna is fine.
[ INFO]: PLEXIL: Pan is fine.
[ INFO]: PLEXIL: Tilt is fine.
[ WARN]: Fault in ANTENNA: TiltFault
[ WARN]: Fault in SYSTEM: PAN_TILT_EXECUTION_ERROR
[ WARN]: PLEXIL: Antenna fault present.
[ INFO]: PLEXIL: Pan is fine.
[ WARN]: PLEXIL: Tilt fault present.
[ WARN]: PLEXIL: Antenna fault present.
[ INFO]: PLEXIL: Pan is fine.
[ WARN]: PLEXIL: Tilt fault present.
[ WARN]: PLEXIL: Antenna fault present.
[ INFO]: PLEXIL: Pan is fine.
[ WARN]: PLEXIL: Tilt fault present.
[ WARN]: Fault in ANTENNA: PanFault
[ WARN]: PLEXIL: Antenna fault present.
[ WARN]: PLEXIL: Pan fault present.
[ WARN]: PLEXIL: Tilt fault present.
[ WARN]: Resolved fault in ANTENNA: TiltFault
[ WARN]: PLEXIL: Antenna fault present.
[ WARN]: PLEXIL: Pan fault present.
[ INFO]: PLEXIL: Tilt is fine.
[ WARN]: PLEXIL: Antenna fault present.
[ WARN]: PLEXIL: Pan fault present.
[ INFO]: PLEXIL: Tilt is fine.
[ WARN]: PLEXIL: Antenna fault present.
[ WARN]: PLEXIL: Pan fault present.
[ INFO]: PLEXIL: Tilt is fine.
[ WARN]: PLEXIL: Antenna fault present.
[ WARN]: PLEXIL: Pan fault present.
[ INFO]: PLEXIL: Tilt is fine.
[ WARN]: Resolved fault in ANTENNA: PanFault
[ WARN]: Resolved fault in SYSTEM: PAN_TILT_EXECUTION_ERROR
[ INFO]: PLEXIL: Antenna is fine.
```
### Notes

- This is a scaled down version of @pthangeda's original work, which at present is not achievable due to the next point.
- Even though pan and tilt faults can now be detected separately, this is of limited use because Pan and Tilt can no longer be commanded separately as they were before.  We plan to bring back this feature, at which time @pthangeda's fault handling example plan can be restored.

### CODE REVIEW

**Comment by kmdalal** [comment]
Overall this looks reasonable and a nice improvement in PLEXIL logic.  The key thing needing change is the simulator side, to comply with the OWLAT unification.  This would impact OwInterface but perhaps not the PLEXIL plan.   I will create a Jira for the unified pan/tilt telemetry, if it doesn't already exist.

Another improvement to the PLEXIL, specifically FaultHandlingPattern7, would be to command only with the established ranges: [-3.2, 3.2] for pan, and [-pi/2, pi/2] for tilt.  I've been working on implementing these range limits in the simulator.

**Comment by kmdalal** [comment]
This branch does not build (using of course the corresponding branch in ow_simulator) and overall strikes me as incomplete work.  E.g. the pan/tilt fault values are not the unified ones, and there are aspects of both the old and new implementations in place.  It also needs to be brought up to date with noetic-devel.

File: ow_plexil/src/plexil-adapter/OwInterface.h [hunk 1]
```diff
@@ -252,9 +254,9 @@ class OwInterface : public PlexilInterface
   };
 
   FaultMap64 m_panTiltErrors = {
-    {"PAN_JOINT_LOCKED", std::make_pair(
+    {"PanFault", std::make_pair(
       owl_msgs::PanTiltFaultsStatus::PAN_JOINT_LOCKED, false)},
-    {"TILT_JOINT_LOCKED", std::make_pair(
+    {"TiltFault", std::make_pair(
```
Comment (anthonykoutroulis) [inline ow_plexil/src/plexil-adapter/OwInterface.h:259]:
```suggestion
    {"TILT_JOINT_LOCKED", std::make_pair(
```
This reverts a telemetry unification update, probably not intended?

**Comment by anthonykoutroulis** [comment]
Test works as described. I only commented on the error messages as they look like an unintended reversion from a merge, but otherwise the code looks fine.

**Comment by kmdalal** [comment]
> Test works as described. I only commented on the error messages as they look like an unintended reversion from a merge, but otherwise the code looks fine.

The map key renames were intentional, but based on an earlier version of the code in which I was using the map key as the PLEXIL lookup name.  I don't see any dependencies on these names, so I thought to make them more useful.  In any case, I reverted them for consistency.

**Review by anthonykoutroulis — APPROVED** [review]
This all looks good to merge.

**Pull request merged**